README.md

```diff
@@ -4,55 +4,146 @@
-- `tests` - this uses [pytest](https://docs.pytest.org/en/stable/) to execute the automated tests in `test/`. There are currently no automated tests written for this repository however.
+Add:
 
-## API Configuration
-In `operationsgateway_api/`, there is an example config file (`config.yml.example`). Copy this example so `config.yml` exists in the same directory level and edit the configuration as needed for your system. There is also a logging configuration file (`logging.ini.example`). Follow the same procedure as the config file as to set up the logging.
+```bash
+[mongodb-org-7.0]
+name=MongoDB Repository
+baseurl=https://repo.mongodb.org/yum/redhat/8/mongodb-org/7.0/x86_64/
+gpgcheck=1
+enabled=1
+gpgkey=https://pgp.mongodb.com/server-7.0.asc
+```
+
+Then:
+
+```bash
+# Install MongoDB
+sudo yum install mongodb-org
+# Start it
+sudo systemctl start mongod
+# Enable it to start on boot
+sudo systemctl enable mongod
+```
 
-## Authentication Configuration
+## Authentication
 
-The authentication system uses JSON Web Tokens (JWTs) which require a private and public key pair to encrypt and decrypt the tokens. The keys can be generated by navigating to a directory and typing the command:
+The authentication system uses JSON Web Tokens (JWTs), which require a private and public key pair to encrypt and decrypt the tokens.
+
+The keys can be generated by navigating to a directory and typing the command:
 
 ```bash
+# When prompted for the location path, enter `id_rsa` to create the keys in the current directory rather than in your home directory.
+# Press enter twice when prompted for the password so as not to set one.
 ssh-keygen -b 2048 -t rsa
 ```
 
-Enter the path for the file as `id_rsa` to create the keys in the current directory rather than in your home directory.
+Then, edit the ```private_key_path``` and ```public_key_path``` settings in the ```auth``` section of the ```config.yml``` file to reflect the location where these keys have been created.
+
+## API Configuration
+
+In `operationsgateway_api/`, there are several example configuration files which need to be reviewed and renamed, removing the `.example` part of the filename.
+
+- `config.yml.example`
+- `logging.ini.example`
+- `maintenance.json.example`
+- `scheduled_maintenance.json.example`
+
+## Test Data
+
+A script has been created to set up the API with some test data and test users. This script also comes with a configuration file that needs to be reviewed and renamed at `util/realistic_data/config.yml.example`.
+
+Once the appropriate configuration has been entered, run the script using Poetry:
+
+```bash
+poetry run python util/realistic_data/ingest_echo_data.py
+```
+
+More details about this script can be found in `docs/developer/test_data.md`
+
+## Starting the API
+
+```bash
+poetry run python operationsgateway_api/src/main.py
```

Comment:
I think Viktor mentioned to me that the current recommended approach is to use [`fastapi run`](https://fastapi.tiangolo.com/deployment/manually/?h=run#use-the-fastapi-run-command), so for us:
```bash
poetry run fastapi run operationsgateway_api/src/main.py
```
Having tried it just now I get more logging and some fancy formatting, but it stops early because my code has syntax errors as I'm mid-development - so I'm not 100% on what would change if we used it, but it might be worth considering?

Reply:
Good spot. I feel this needs investigating a bit more to fully understand what it does. I've created an issue [here](https://stfc.atlassian.net/browse/DSEGOG-385)

Reply:
Yeah I suppose it might also be of interest in the ansible repo, I don't know how that actually runs things for deployment. There's also `fastapi dev` which does the hot reloads when the source code changes.